app/views/admin/topics/_ticket.html.erb

```diff
@@ -6,7 +6,7 @@
 
   <div id="ticket-controls" class="col-md-8 col-xs-12 col-lg-8">
     <h2>
-      <%= render 'admin/topics/ticket_nav_dropdown' %> #<%= @topic.id %>- <%= @topic.name -%>
+      <%= render 'admin/topics/ticket_nav_dropdown' %> #<%= @topic.id %>- <%= best_in_place @topic, :name, as: :input, activator: :span -%>&nbsp;<span class="btn" id="btn_edit_topic_name">Edit</span>
```

Comment:
Currently this code is passed through the `TopicsController#update` which doesn't check whether the user completing the request is an admin. (This means that any user on the system could change the name of any topic.)

It would be best if the request goes through `Admin::TopicsController#update`, this can easily be done by moving your changes to `TopicsController` over to `Admin::TopicsController` and then changing this line to include `url: admin_topic_path(@topic)`